src/index.ts

```diff
@@ -0,0 +1,140 @@
+const allowedMethods = ['GET', 'HEAD', 'OPTIONS'];
+const defaultCorsHeaders = {
+	allow: allowedMethods.join(', '),
+	'Access-Control-Allow-Headers': '*',
+	'Access-Control-Allow-Methods': allowedMethods.join(', '),
+	'Access-Control-Allow-Origin': '*',
+};
+
+function makeError(code: number, error: String, description: String): Response {
+	return new Response(JSON.stringify({ error, description }), {
+		status: code,
+		headers: {
+			...defaultCorsHeaders,
+			'content-type': 'application/json',
+		},
+	});
+}
+
+interface UrlData {
+	projectId: string;
+	versionId: string;
+}
+
+function extractUrlData(path: string): UrlData | null {
+	// Pattern: data/:hash/versions/:version/:file
+	const parts = path.split('/');
+	console.log(parts);
+
+	const hashIndex = parts.findIndex((part) => part === 'data') + 1;
+	const versionIndex = parts.findIndex((part) => part === 'versions') + 1;
+
+	console.log(hashIndex);
+	console.log(versionIndex);
+
+	if (hashIndex === -1 || versionIndex === -1) {
+		return null;
+	}
+
+	const projectId = parts[hashIndex];
+	const versionId = parts[versionIndex];
+
+	return { projectId, versionId };
+}
+
+async function countDownload(request: Request, env: Env, urlData: UrlData) {
+	console.log(`Attempting to count download of project ID ${urlData.projectId} with version ID ${urlData.versionId}`);
+
+	const url = `${env.LABRINTH_URL}admin/_count-download`;
+
+	console.log(`url: ${url}`);
```

Comment:
do we care about console logs here?

Reply:
logs in workers are fine--they're just stored for a small bit and could be hooked into to see if there's anything fishy or errors going on.